src/api/dataops/dataset.spec.ts

```diff
@@ -79,4 +78,23 @@ describe("Dataset class", () => {
     });
   });
 
+  describe("On detach", () => {
+    it("should return the query by passing a filter", () => {
+      const dataset = new Dataset("test", createMockFor(RequestExecuter));
+      const query = dataset.detach(
+        faker.random.word(),
+        getRandomFilteringCriteria(),
+      );
+
+      expect(query instanceof ActionQuery).toBeTruthy();
```

Comment:
I think we need a test that `detach()` returns an ActionQuery with ActionType = `detach`, don't we?

Reply:
done!